Can't install it with rustc 1.86.0

When I do `cargo install ripunzip` it fails with:

```
...

   Compiling progress-streams v1.1.0
   Compiling ranges v0.4.0
   Compiling wildmatch v2.4.0
   Compiling zstd v0.13.3
   Compiling ripunzip v2.0.1
error[E0107]: missing generics for struct `ZipFile`
   --> /home/yehor/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ripunzip-2.0.1/src/unzip/mod.rs:388:11
    |
388 |     file: ZipFile,
    |           ^^^^^^^ expected 1 generic argument
    |
note: struct defined here, with 1 generic parameter: `R`
   --> /home/yehor/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/zip-2.6.1/src/read.rs:252:12
    |
252 | pub struct ZipFile<'a, R: Read> {
    |            ^^^^^^^     -
help: add missing generic argument
    |
388 |     file: ZipFile<R>,
    |                  +++

error[E0107]: missing generics for struct `ZipFile`
   --> /home/yehor/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ripunzip-2.0.1/src/unzip/mod.rs:405:15
    |
405 |     mut file: ZipFile,
    |               ^^^^^^^ expected 1 generic argument
    |
note: struct defined here, with 1 generic parameter: `R`
   --> /home/yehor/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/zip-2.6.1/src/read.rs:252:12
    |
252 | pub struct ZipFile<'a, R: Read> {
    |            ^^^^^^^     -
help: add missing generic argument
    |
405 |     mut file: ZipFile<R>,
    |                      +++

error[E0107]: missing generics for struct `ZipFile`
   --> /home/yehor/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ripunzip-2.0.1/src/unzip/mod.rs:352:31
    |
352 |                     let file: ZipFile = match &options.password {
    |                               ^^^^^^^ expected 1 generic argument
    |
note: struct defined here, with 1 generic parameter: `R`
   --> /home/yehor/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/zip-2.6.1/src/read.rs:252:12
    |
252 | pub struct ZipFile<'a, R: Read> {
    |            ^^^^^^^     -
help: add missing generic argument
    |
352 |                     let file: ZipFile<R> = match &options.password {
    |                                      +++

error[E0107]: missing generics for struct `ZipFile`
   --> /home/yehor/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/ripunzip-2.0.1/src/unzip/mod.rs:380:15
    |
380 |     let file: ZipFile = match password {
    |               ^^^^^^^ expected 1 generic argument
    |
note: struct defined here, with 1 generic parameter: `R`
   --> /home/yehor/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/zip-2.6.1/src/read.rs:252:12
    |
252 | pub struct ZipFile<'a, R: Read> {
    |            ^^^^^^^     -
help: add missing generic argument
    |
380 |     let file: ZipFile<R> = match password {
    |                      +++

For more information about this error, try `rustc --explain E0107`.
error: could not compile `ripunzip` (lib) due to 4 previous errors
warning: build failed, waiting for other jobs to finish...
error: failed to compile `ripunzip v2.0.1`, intermediate artifacts can be found at `/tmp/cargo-install1jiF6x`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -36,7 +36,7 @@ reqwest = { version = "0.11.13", features = ["blocking"] }
 tempfile = "3.3.0"
 thiserror = "1.0.37"
 wildmatch = "2.1.1"
-zip = "2.3"
+zip = "2.6"
 
 [dev-dependencies]
 hexdump = "0.1.1"

--- a/src/unzip/mod.rs
+++ b/src/unzip/mod.rs
@@ -349,7 +349,7 @@ fn unzip_serial_or_parallel<'a, T: Read + Seek + 'a>(
                 .into_iter()
                 .map(|name| {
                     let myzip: &mut zip::ZipArchive<T> = &mut get_ziparchive_clone();
-                    let file: ZipFile = match &options.password {
+                    let file: ZipFile<T> = match &options.password {
                         None => myzip.by_name(&name)?,
                         Some(string) => myzip.by_name_decrypt(&name, string.as_bytes())?,
                     };
@@ -377,15 +377,15 @@ fn extract_file_by_index<'a, T: Read + Seek + 'a>(
     directory_creator: &DirectoryCreator,
 ) -> Result<(), anyhow::Error> {
     let myzip: &mut zip::ZipArchive<T> = &mut get_ziparchive_clone();
-    let file: ZipFile = match password {
+    let file: ZipFile<T> = match password {
         None => myzip.by_index(i)?,
         Some(string) => myzip.by_index_decrypt(i, string.as_bytes())?,
     };
     extract_file(file, output_directory, progress_reporter, directory_creator)
 }
 
-fn extract_file(
-    file: ZipFile,
+fn extract_file<R: Read>(
+    file: ZipFile<R>,
     output_directory: &Option<PathBuf>,
     progress_reporter: &dyn UnzipProgressReporter,
     directory_creator: &DirectoryCreator,
@@ -401,8 +401,8 @@ fn extract_file(
 }
 
 /// Extracts a file from a zip file.
-fn extract_file_inner(
-    mut file: ZipFile,
+fn extract_file_inner<R: Read>(
+    mut file: ZipFile<R>,
     output_directory: &Option<PathBuf>,
     progress_reporter: &dyn UnzipProgressReporter,
     directory_creator: &DirectoryCreator,

